### DISCUSSION TITLE
Regarding Fuzzing using AFL

### DISCUSSION THREAD

**Discussion by aswintr1998** [DMTF/libspdm#2524] [General]
I am a newbie here. Here I have seen the instruction to perform fuzzing of libspdm library using AFL. I have seen a bash script "fuzzing_AFL.sh" which is executing the same. The input to the fuzzer was the ".raw" formatted files in the "Seeds" folder. Currently I am not able to open the ".raw" input files. How can I open and analyze the  ".raw"
![Raw_data](https://github.com/DMTF/libspdm/assets/144205188/8abd7b84-67ec-4698-abeb-51c56fa0451f)
 input data?  Do we have a sample code of the compiled binary?

**Comment by steven-bellock** [comment]
Those are raw binary files so you'll need a hex editor to view them. https://hexed.it/ is an online one. Notepad++ has a hex editor plugin as well.

**Reply by aswintr1998** [reply]
Thank you. How do we generate these raw binary files?

**Reply by steven-bellock** [reply]
@Zhiqiang520 and/or @Xiaohanjlll probably know.

**Reply by aswintr1998** [reply]
@Zhiqiang520 / @Xiaohanjlll could you kindly help me on the above-mentioned query?

**Comment by Xiaohanjlll** [comment]
```
    spdm_get_version_request_t libspdm_get_version_request = {
        {
            SPDM_MESSAGE_VERSION_10,
            SPDM_GET_VERSION,
        },
    };
 
    FILE *fp;
    fp = fopen("./version.raw", "w+");
    fwrite(&libspdm_get_version_request, sizeof(libspdm_get_version_request), 1, fp);
    fclose(fp);
```

@aswintr1998
